File: 0405-convert-a-number-to-hexadecimal/0405-convert-a-number-to-hexadecimal.py
class Solution:
    def toHex(self, num):
        if num == 0:
            return "0"

        digits = "0123456789abcdef"
        result = ""

        # Convert negative number to 32-bit two's complement
        if num < 0:
            num += 2**32

        while num > 0:
            result = digits[num & 15] + result
            num >>= 4

        return result

File: 0405-convert-a-number-to-hexadecimal/0405-convert-a-number-to-hexadecimal.cpp
class Solution {
public:
    string toHex(int num) {
        if (num == 0)
            return "0";

        string hex = "0123456789abcdef";
        string result = "";

        unsigned int n = num;

        while (n > 0) {
            int digit = n & 15;   // Get last 4 bits
            result += hex[digit];
            n >>= 4;              // Remove last 4 bits
        }

        reverse(result.begin(), result.end());

        return result;
    }
};